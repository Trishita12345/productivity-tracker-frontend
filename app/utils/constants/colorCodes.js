export const COLORS = {
  LIGHT_GREY: '#F5F5F5',
  GREY: '#C6C6C6',
  DARK_GREY: '#87797F',
  VIOLET: '#70367C',
  ORANGE: '#FD7E50',
  SKY: '#95D0D5',
  DARK_VIOLET: '#301137',
  GREEN: '#6DB33F',
  YELLOW: '#DCD41D',
  RED: '#E41515',
  WHITE: '#FFFAFF',
  BLUE: '#4E63B2',
};
