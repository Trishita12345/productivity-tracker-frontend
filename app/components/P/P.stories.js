import React from 'react';
import P from '.';
import { VARIANTS } from '../../utils/constants/variants';

export default {
  title: 'form/P',
  component: P,
  args: {
    label: 'Label',
  },
};

const Template = args => <P {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  variant: VARIANTS.PRIMARY,
};

export const Secondary = Template.bind({});
Secondary.args = {
  variant: VARIANTS.SECONDARY,
};

export const Tertiary = Template.bind({});
Tertiary.args = {
  variant: VARIANTS.TERTIARY,
};
