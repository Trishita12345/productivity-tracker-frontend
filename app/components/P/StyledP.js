import styled from 'styled-components';
import { COLORS } from '../../utils/constants/colorCodes';

const textColor = {
  primary: COLORS.DARK_VIOLET,
  secondary: COLORS.WHITE,
  tertiary: COLORS.DARK_GREY,
};

const StyledP = styled.p`
  font-family: 'Roboto Slab';
  font-style: normal;
  font-size: 1em;
  line-height: 24px;
  color: ${({ variant }) => textColor[variant]};
  margin-bottom: 0.25em;
`;

export default StyledP;
