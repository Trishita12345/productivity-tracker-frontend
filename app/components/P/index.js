import React from 'react';
import PropTypes from 'prop-types';

import StyledP from './StyledP';
import { VARIANTS } from '../../utils/constants/variants';

const P = ({ label, variant = VARIANTS.PRIMARY }) => (
  <StyledP variant={variant}>{label}</StyledP>
);

P.propTypes = {
  label: PropTypes.string,
  variant: PropTypes.oneOf([
    VARIANTS.PRIMARY,
    VARIANTS.SECONDARY,
    VARIANTS.TERTIARY,
  ]),
};

export default P;
