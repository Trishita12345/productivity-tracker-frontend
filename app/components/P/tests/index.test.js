/**
 * Testing our P component
 */

import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import P from '../index';
import { VARIANTS } from '../../../utils/constants/variants';
import StyledP from '../StyledP';

configure({ adapter: new Adapter() });

describe('<P />', () => {
  const props = {
    label: 'Label',
  };

  const ComponentRenderer = myProps => {
    const component = shallow(<P {...myProps} />);
    return component;
  };

  it('should P renders', () => {
    const component = ComponentRenderer(props);
    expect(component).toBeDefined();
    expect(component.find(StyledP).text()).toEqual('Label');
  });
  it('should renders primary as default', () => {
    const component = ComponentRenderer(props);
    expect(component.find(StyledP).prop('variant')).toEqual(VARIANTS.PRIMARY);
    expect(component).toMatchSnapshot();
  });
  it('should renders secondary variant', () => {
    const component = ComponentRenderer({
      ...props,
      variant: VARIANTS.SECONDARY,
    });
    expect(component.find(StyledP).prop('variant')).toEqual(VARIANTS.SECONDARY);
    expect(component).toMatchSnapshot();
  });
  it('should renders Tertiary variant', () => {
    const component = ComponentRenderer({
      ...props,
      variant: VARIANTS.TERTIARY,
    });
    expect(component.find(StyledP).prop('variant')).toEqual(VARIANTS.TERTIARY);
    expect(component).toMatchSnapshot();
  });
});
