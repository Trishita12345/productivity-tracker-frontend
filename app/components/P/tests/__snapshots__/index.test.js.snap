// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<P /> should renders Tertiary variant 1`] = `ShallowWrapper {}`;

exports[`<P /> should renders primary as default 1`] = `ShallowWrapper {}`;

exports[`<P /> should renders secondary variant 1`] = `ShallowWrapper {}`;
